client/landing/gutenboarding/hooks/use-typer.ts

```diff
@@ -0,0 +1,84 @@
+/**
+ * External dependencies
+ */
+import { useState } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import { useInterval } from 'lib/interval/use-interval';
+
+type TyperMode = 'TYPING' | 'DELETING';
+
+type AllTyperOptions = {
+	delayBetweenCharacters: number;
+	delayBetweenWords: number;
+};
+
+// if options are passed. At least one of them should be set
+type TyperOptions =
+	| Pick< AllTyperOptions, 'delayBetweenCharacters' >
+	| Pick< AllTyperOptions, 'delayBetweenWords' >
+	| AllTyperOptions;
+
+const defaultOptions: AllTyperOptions = { delayBetweenCharacters: 120, delayBetweenWords: 1200 };
```

Comment:
Use `Options`:

```tsx
const defaultOptions: Options = { delayBetweenCharacters: 120, delayBetweenWords: 1200 };
```

We can also prevent it from being modified:

```tsx
const defaultOptions: Readonly< Options > = { delayBetweenCharacters: 120, delayBetweenWords: 1200 };
```

Finally, I'd probably use CAPITAL_SNAKE_CASE for this constant:

```suggestion
const DEFAULT_OPTIONS: Readonly< Options > = { delayBetweenCharacters: 120, delayBetweenWords: 1200 };
```